### DISCUSSION TITLE
Failed to load libCEC library. One of the CEC library component is missing

### DISCUSSION THREAD

**Discussion by Psyobilin** [awawa-dev/HyperHDR#783] [Q&A] [answered]
My grabber is capable of CEC and hyperHDR tries to activate it but it does not work. It quits with 
Failed to load libCEC library. One of the CEC library component is missing. 
Could not open libcec library
Stopping CEC
I tried to update the cec library but I still have the same problem.
Any ideas?

**Comment by awawa-dev** [comment]
In typical setup the grabber has nothing to do with CEC. If you are using different that RPi system or you are not using our DEB installer then you must investigate what is missing on your own. Maybe installing libcec via apt wold be the easiest way.

**Comment by stockholm11** [comment]
My setup: Raspberry Pi 4, NodeMCU, 2x4 HDMI 2.1 switcher(apple tv + decoder), 5V 20A adapter, 225 led ws2801b, ms2109.

Automatic signal detection is enabled, but HyperHDR does not detect that the Apple TV is off. It lights up the LEDs according to this image.
![shot005510](https://github.com/user-attachments/assets/ae2df7c6-d716-4548-8509-21f3819c7347)
I have to disable the instance every time I turn off the TV.

I have installed the libCEC library and even cec-client multiple times.
I also added these lines to the /boot/firmware/config.txt file:
hdmi_force_hotplug=1
hdmi_driver=2
I added the line /lib/aarch64-linux-gnu to the /etc/ld.so.conf.d/cec.conf file.
However, no matter what I tried, I keep getting the error: "Failed to load libCEC library. One of the CEC library components is missing."
What else should I check or do?

**Reply by rjmgmt** [reply, marked as answer]
You don't need CEC for that. You need to set and calibrate the automatic signal detection setting under the USB capture device settings. Just enable it and hit calibrate when the rainbow is displayed.

**Reply by awawa-dev** [reply]
CEC only works when the RPI HDMI is connected to the TV. Then it should receive the signal whether the TV is on or off. If an HDMI splitter is used for such a connection, it must also support CEC. I have not come across any USB loopback grabber that supports CEC.

**Reply by stockholm11** [reply]
> You don't need CEC for that. You need to set and calibrate the automatic signal detection setting under the USB capture device settings. Just enable it and hit calibrate when the rainbow is displayed.

Thank you very much. My problem has been resolved.
